backend/apps/schemes/api/serializers.py

```
@@ -133,15 +133,44 @@ def validate_description(self, value):
 
 
+class PlanField(serializers.Field):
+    """Custom field to handle plan ID to Plan instance conversion."""
+    
+    def to_internal_value(self, data):
+        if data is None or data == '':
+            return None
+        try:
+            return Plan.objects.get(id=data)
+        except Plan.DoesNotExist:
+            raise serializers.ValidationError(f'Plan with id {data} does not exist.')
```

Ruff (0.14.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)